dbt_cloud_plugin/hooks/dbt_cloud_hook.py

```diff
@@ -68,6 +68,39 @@ def get_run_status(self, run_id):
         status_name = RunStatus.lookup(run['status'])
         return status_name
 
+    def get_run_manifest(self, run_id):
+        """
+        Return the manifest.json from a dbt Cloud run.
+        """
+        dbt_cloud = self.get_conn()
+        return dbt_cloud.get_artifact(run_id, 'manifest.json')
+
+    def get_all_run_results(self, run_id):
+        """
+        Return the results array from run_results.json from a dbt Cloud run,
+        concatenated across all (real) steps.
+        """
+        dbt_cloud = self.get_conn()
+
+        # first, determine the number of steps in this job
+        # it will either be defined in the run or in the job definition
```

Comment:
It's not entirely clear from the API documentation, but reading between the lines a bit: you can override the default steps when you trigger the job to run via the API. If you do that, then steps will be returned from the run, otherwise they're in the job definition.